src/Sparrow/Utils/NativeMemoryCleaner.cs

```diff
@@ -81,7 +80,18 @@ public void CleanNativeMemory(object state)
 
         public void Dispose()
         {
-            _timer.Dispose();
+#if !NETSTANDARD1_3
+            using (var waitHandle = new ManualResetEvent(false))
+            {
+                if (_timer.Dispose(waitHandle))
+                {
+                    waitHandle.WaitOne();
+                }
+            }
+#else
+            Monitor.Enter(_lock); // prevent from running the callback _after_ dispose
```

Comment:
Not a good idea. If you are running into this, it will hang the timer thread.

Reply:
The timer thread (in the callback method) just calls:
`Monitor.TryEnter(_lock, ref lockTaken);` 

https://github.com/ravendb/ravendb/blob/789e53205e9d1c46e725dd60da02b84bc955f3cd/src/Sparrow/Utils/NativeMemoryCleaner.cs#L28-L30

So it won't hang it.

Reply:
what happens if Dispose is called twice?